CMakeLists.txt

```diff
@@ -160,32 +160,36 @@ install(TARGETS SDL
 )
 
 if(SDL12DEVEL)
+  set(SDL_VERSION "${SDL12_COMPAT_VERSION_STR}")
+  set(SDL_CFLAGS "-D_GNU_SOURCE=1 -D_REENTRANT")
+  set(SDL_RLD_FLAGS "")  # !!! FIXME: this forces rpath, which we might want?
+  set(SDL_LIBS "-lSDL")
+  set(SDL_STATIC_LIBS "-lm -ldl -lpthread")
+
   install(DIRECTORY include/ DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
   )
   configure_file(sdl12_compat.pc.in sdl12_compat.pc @ONLY)
   install(FILES ${CMAKE_BINARY_DIR}/sdl12_compat.pc
     DESTINATION ${CMAKE_INSTALL_LIBDIR}/pkgconfig
   )
 
-  # !!! FIXME: lots of these sdl-config vars probably need more customization.
-
   # !!! FIXME: do we _want_ static builds?
   set(ENABLE_STATIC_TRUE "")
   set(ENABLE_STATIC_FALSE "#")
   set(ENABLE_SHARED_TRUE "")
   set(ENABLE_SHARED_FALSE "#")
 
-  set(SDL_VERSION "${SDL12_COMPAT_VERSION_STR}")
-  set(SDL_CFLAGS "-D_GNU_SOURCE=1 -D_REENTRANT")
-  set(SDL_RLD_FLAGS "")  # !!! FIXME: this forces rpath, which we might want?
-  set(SDL_LIBS "-lSDL")
-  set(SDL_STATIC_LIBS "-lm -ldl -lpthread")
-  set(prefix ${CMAKE_INSTALL_PREFIX})
-  set(exec_prefix "\${prefix}")
-  set(libdir "\${exec_prefix}/lib${LIB_SUFFIX}")
-  set(bindir "\${exec_prefix}/bin")
-  set(includedir "\${prefix}/include")
   configure_file("${CMAKE_SOURCE_DIR}/sdl-config.in" "${CMAKE_BINARY_DIR}/sdl-config" @ONLY)
   install(PROGRAMS "${CMAKE_BINARY_DIR}/sdl-config" DESTINATION bin)
+
+  set(SOEXT ${CMAKE_SHARED_LIBRARY_SUFFIX})
+  get_target_property(SONAME SDL OUTPUT_NAME)
+  if(NOT ("${SONAME}" EQUAL "SDL"))
```

Comment:
I guess this needs guarding with something like `if(NOT (WIN32 OR CYGWIN))`

Reply:
Well it could be an `if(UNIX AND NOT ANDROID)` given the `OUTPUT_NAME` logic in CMakeLists.txt:54 but I think it's better to avoid duplicating that logic.

Reply:
@icculus, this is yours

Reply:
I guess wrap it in (UNIX AND NOT ANDROID) for now.